strings/my_vsnprintf.c

```diff
@@ -567,40 +581,47 @@ static char *process_args(CHARSET_INFO *cs, char *to, char *end,
       case 'X':
       case 'o':
       case 'p':
-      {
-        /* Integer parameter */
-        longlong larg;
-        length= (print_arr[i].flags & LENGTH_ARG)
-          ? (size_t)args_arr[print_arr[i].length].longlong_arg
-          : print_arr[i].length;
-
-        larg = args_arr[print_arr[i].arg_idx].longlong_arg;
-        to= process_int_arg(to, end, length, larg, print_arr[i].arg_type,
-                            print_arr[i].flags);
-        break;
-      }
       case 'M':
       {
-        longlong larg;
-        const char *real_end;
-
-        width= (print_arr[i].flags & WIDTH_ARG)
-          ? (size_t)args_arr[print_arr[i].width].longlong_arg
-          : print_arr[i].width;
-
-        real_end= MY_MIN(to + width, end);
-
-        larg = args_arr[print_arr[i].arg_idx].longlong_arg;
-        to= process_int_arg(to, real_end, 0, larg, 'd', print_arr[i].flags);
-        if (real_end - to >= 3)
+        /* Integer parameter */
+        longlong larg= args_arr[print_arr[i].arg_idx].longlong_arg;
+        my_bool suffix_e= print_arr[i].arg_type == 'M';
+        if (print_arr[i].arg_type == 'u')
```

Comment:
Really minor potential optimisation (and readability improvement):

Before the switch, do something like:

```c
char arg_type= print_arr[i].arg_type;
```

And then you can use `arg_type` everywhere.

Note: you don't have to make this change.

Reply:
How’s this?